src/libraries/Native/Unix/System.Security.Cryptography.Native.Android/pal_x509.c

```diff
@@ -0,0 +1,458 @@
+    loc[exts] = critical
+        ? (*env)->CallObjectMethod(env, cert, g_X509CertGetCriticalExtensionOIDs)
+        : (*env)->CallObjectMethod(env, cert, g_X509CertGetNonCriticalExtensionOIDs);
+    ON_EXCEPTION_PRINT_AND_GOTO(cleanup);
+
+    // Iterator<string> iter = collection.iterator();
+    // while (iter.hasNext()) {
+    //     string oid = iter.next();
+    //     byte[] data = cert.getExtensionValue(oid);
+    //     cb(oid, oid.length, data, data.length, /*critical*/ true, context);
+    // }
+    loc[iter] = (*env)->CallObjectMethod(env, loc[exts], g_SetIterator);
+    ON_EXCEPTION_PRINT_AND_GOTO(cleanup);
+    jboolean hasNext = (*env)->CallBooleanMethod(env, loc[iter], g_IteratorHasNext);
+    ON_EXCEPTION_PRINT_AND_GOTO(cleanup);
+    while (hasNext)
+    {
+        INIT_LOCALS(locLoop, oid, data)
+        const char *oidPtr = NULL;
+        jbyte *dataPtr = NULL;
+
+        locLoop[oid] = (*env)->CallObjectMethod(env, loc[iter], g_IteratorNext);
+        ON_EXCEPTION_PRINT_AND_GOTO(loop_cleanup);
+        jsize oidLen = (*env)->GetStringUTFLength(env, locLoop[oid]);
+        oidPtr = (*env)->GetStringUTFChars(env, locLoop[oid], NULL);
+
+        locLoop[data] = (*env)->CallObjectMethod(env, cert, g_X509CertGetExtensionValue, locLoop[oid]);
+        ON_EXCEPTION_PRINT_AND_GOTO(loop_cleanup);
+        jsize dataLen = (*env)->GetArrayLength(env, locLoop[data]);
+        dataPtr = (*env)->GetByteArrayElements(env, locLoop[data], NULL);
+
+        // X509Certificate.getExtensionValue returns the full DER-encoded data.
+        cb(oidPtr, oidLen, (const uint8_t *)dataPtr, dataLen, critical, context);
+
+        hasNext = (*env)->CallBooleanMethod(env, loc[iter], g_IteratorHasNext);
+        if (CheckJNIExceptions(env))
+            hasNext = false;
+
+    loop_cleanup:
+        if (oidPtr != NULL)
+            (*env)->ReleaseStringUTFChars(env, locLoop[oid], oidPtr);
+
+        if (dataPtr != NULL)
+            (*env)->ReleaseByteArrayElements(env, locLoop[data], dataPtr, JNI_ABORT);
+
+        RELEASE_LOCALS(locLoop, env)
+    }
+
+cleanup:
+    RELEASE_LOCALS(loc, env)
+}
+
+int32_t AndroidCryptoNative_X509EnumExtensions(jobject /*X509Certificate*/ cert, EnumX509ExtensionsCallback cb, void *context)
+{
+    assert(cert != NULL);
+    JNIEnv *env = GetJNIEnv();
+
+    // [TODO] We're skipping over any errors - do we care to propagate?
+    EnumExtensions(env, cert, true /*critical*/, cb, context);
+    EnumExtensions(env, cert, false /*critical*/, cb, context);
+    return SUCCESS;
+}
+
+int32_t AndroidCryptoNative_X509FindExtensionData(jobject /*X509Certificate*/ cert, const char *oid, uint8_t *buf, int32_t len)
+{
+    LOG_ERROR("Not yet implemented");
+
+    // byte[] data = cert.getExtensionValue(oid);
+    // return data.length;
+    return BUFFER_FAIL;
+}
+
+jobject /*X509CRL*/ AndroidCryptoNative_DecodeX509Crl(const uint8_t *buf, int32_t len)
```

Comment:
We may not need this on Android, at least not yet.  I'm hoping that we can be a bit more hands-off on how revocation gets handled than with OpenSSL (but still be successful)